bench/irmin-pack/tree.ml

```diff
@@ -0,0 +1,427 @@
+    (config, result)
+end
+
+module Bench_inodes_32 = Bench_suite (Conf)
+
+module Bench_inodes_2 = Bench_suite (struct
+  let entries = 2
+  let stable_hash = 5
+end)
+
+type mode_elt = [ `Read_trace | `Chains | `Large ]
+
+type suite_elt = {
+  mode : mode_elt;
+  speed : [ `Quick | `Slow ];
+  inode_config : [ `Entries_32 | `Entries_2 ];
+  run : config -> (config * Benchmark.result) Lwt.t;
+}
+
+let suite : suite_elt list =
+  [
+    {
+      mode = `Read_trace;
+      speed = `Quick;
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run_read_trace ~quick:();
+    };
+    {
+      mode = `Read_trace;
+      speed = `Slow;
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run_read_trace;
+    };
+    {
+      mode = `Chains;
+      speed = `Quick;
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run ~mode:`Chains;
+    };
+    {
+      mode = `Chains;
+      speed = `Slow;
+      inode_config = `Entries_2;
+      run = Bench_inodes_2.run ~mode:`Chains;
+    };
+    {
+      mode = `Large;
+      speed = `Quick;
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run ~mode:`Large;
+    };
+    {
+      mode = `Large;
+      speed = `Slow;
+      inode_config = `Entries_2;
+      run = Bench_inodes_2.run ~mode:`Large;
+    };
+  ]
+
+let pp_inode_config fmt = function
+  | `Entries_2 -> Format.fprintf fmt "[2, 5]"
+  | `Entries_32 -> Format.fprintf fmt "[32, 256]"
+
+let pp_config b config fmt () =
+  match b.mode with
+  | `Read_trace ->
+      let stats = Bench_inodes_32.Trees_trace.get_stats () in
+      Format.fprintf fmt
+        "Tezos_log mode on inode config %a: %d commits (%d adds; %d removes; \
+         %d finds; %d mems; %d mem_tree; %d copies)"
+        pp_inode_config b.inode_config config.ncommits_trace stats.adds
+        stats.removes stats.finds stats.mems stats.mem_tree stats.copies
+  | `Chains ->
+      Format.fprintf fmt
+        "Chain trees mode on inode config %a: %d commits, each consisting of \
+         %d chains of depth %d"
+        pp_inode_config b.inode_config config.ncommits config.nchain_trees
+        config.depth
+  | `Large ->
+      Format.fprintf fmt
+        "Large trees mode on inode config %a: %d commits, each consisting of \
+         %d large trees of %d entries"
+        pp_inode_config b.inode_config config.ncommits config.nlarge_trees
+        config.width
+
+let main ncommits ncommits_trace operations_file quick depth width nchain_trees
+    nlarge_trees =
+  let config =
+    {
+      ncommits;
+      ncommits_trace;
+      operations_file;
+      root = "test-bench";
+      quick;
+      depth;
+      width;
+      nchain_trees;
+      nlarge_trees;
+    }
+  in
+  Printexc.record_backtrace true;
+  Random.self_init ();
+  FSHelper.rm_dir config.root;
+  let suite =
+    if config.quick then List.filter (fun b -> b.speed = `Quick) suite
+    else (* only run first element of the suite when quick *) List.tl suite
```

Comment:
something is wrong here, i'm not sure what you meant to do :)

Reply:
I clarified a bit that comment.